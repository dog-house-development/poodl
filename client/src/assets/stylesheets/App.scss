@import './layout/header.scss';
@import './layout/footer.scss';
@import './buttons.scss';
@import './field.scss';
