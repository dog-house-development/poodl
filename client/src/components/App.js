import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import jwt_decode from 'jwt-decode';
import setAuthToken from '../utils/setAuthToken';

import { setCurrentAdmin, logoutAdmin } from '../actions/authActions';
import { Provider } from 'react-redux';
import store from '../store';

// layout
import Navbar from './layout/Navbar';
import Footer from './layout/Footer';

// auth
import Register from './pages/auth/Register';
import Login from './pages/auth/Login';

// basic
import Dashboard from './pages/dashboard/Dashboard';
import Landing from './pages/Landing';
import PageNotFound from './pages/PageNotFound';

// view all pages
import ViewAllAdmins from './pages/views/ViewAllAdmins';
import ViewAllVolunteers from './pages/views/ViewAllVolunteers';
import ViewAllMembers from './pages/views/ViewAllMembers';

// profile pages
import AdminProfile from './pages/views/AdminProfile';
import VolunteerProfile from './pages/views/VolunteerProfile';
import MemberProfile from './pages/views/MemberProfile';

import PrivateRoute from './private-route/PrivateRoute';

import '../assets/stylesheets/App.scss';

// Check for token to keep admin logged in
if (localStorage.jwtToken) {
    // Set auth token header auth
    const token = localStorage.jwtToken;
    setAuthToken(token);
    // Decode token and get admin info and exp
    const decoded = jwt_decode(token);
    // Set admin and isAuthenticated
    store.dispatch(setCurrentAdmin(decoded));
    // Check for expired token
    const currentTime = Date.now() / 1000; // to get in milliseconds
    if (decoded.exp < currentTime) {
        // Logout admin
        store.dispatch(logoutAdmin());

        // Redirect to login
        window.location.href = './login';
    }
}
class App extends Component {
    render() {
        return (
            <Provider store={store}>
                <Router>
                    <div className="App wrapper">
                        <Navbar />
                        <Switch>
                            <Route exact path="/" component={Landing} />
                            <Route exact path="/login" component={Login} />
                            <PrivateRoute exact path="/register" component={Register} />
                            <PrivateRoute exact path="/dashboard" component={Dashboard} />
                            {/* View all pages */}
                            <PrivateRoute exact path="/admins" component={ViewAllAdmins} />
                            <PrivateRoute exact path="/volunteers" component={ViewAllVolunteers} />
                            <PrivateRoute exact path="/members" component={ViewAllMembers} />
                            {/* Profile pages */}
                            <PrivateRoute exact path="/admin/:id" component={AdminProfile} />
                            <PrivateRoute exact path="/volunteer/:id" component={VolunteerProfile} />
                            <PrivateRoute exact path="/member/:id" component={MemberProfile} />
                            {/* PageNotFound route must stay at bottom */}
                            <Route component={PageNotFound} />
                        </Switch>
                        <Footer />
                    </div>
                </Router>
            </Provider>
        );
    }
}
export default App;
