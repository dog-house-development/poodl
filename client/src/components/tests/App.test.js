import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import App from '../App';

configure({ adapter: new Adapter() });

describe('Navbar tests', () => {
    let wrapper, instance;
    const setInstanceAndWrapper = () => {
        wrapper = shallow(<App />);
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('render', () => {
        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
