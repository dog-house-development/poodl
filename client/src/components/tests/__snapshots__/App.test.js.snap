// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests render should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <div
      className="App wrapper"
    >
      <withRouter(Connect(Navbar)) />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/login"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/dashboard"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/admins"
          restrictAccess="Volunteer"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/admins/register"
          restrictAccess="Volunteer"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/admins/:id"
          restrictAccess="Volunteer"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/volunteers"
          restrictAccess="Volunteer"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/volunteers/register"
          restrictAccess="Volunteer"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/volunteers/:id"
          restrictAccess="Volunteer"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/members"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/members/register"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/members/:id"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/activities"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/activities/add"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/activities/:id"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/services/:id"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/member-check-in"
        />
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          path="/reports"
        />
        <Route
          component={[Function]}
        />
      </Switch>
      <Footer />
    </div>
  </BrowserRouter>
</Provider>
`;
